nebula_decoders/include/nebula_decoders/nebula_decoders_hesai/decoders/pandar_xt16.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "nebula_decoders/nebula_decoders_hesai/decoders/hesai_packet.hpp"
+#include "nebula_decoders/nebula_decoders_hesai/decoders/hesai_sensor.hpp"
+
+#include <vector>
+
+namespace nebula::drivers
+{
+
+namespace hesai_packet
+{
+
+#pragma pack(push, 1)
+
+struct TailXT16
+{
+  uint8_t reserved1[10];
+  uint8_t return_mode;
+  uint16_t motor_speed;
+  DateTime<1900> date_time;
+  uint32_t timestamp;
+  uint8_t factory_information;
+};
+
+struct PacketXT16 : public PacketBase<8, 16, 2, 100>
+{
+  using body_t = Body<Block<Unit4B, PacketXT16::n_channels>, PacketXT16::n_blocks>;
+  Header12B header;
+  body_t body;
+  TailXT16 tail;
+  uint32_t udp_sequence;
+};
+
+#pragma pack(pop)
+
+}  // namespace hesai_packet
+
+class PandarXT16 : public HesaiSensor<hesai_packet::PacketXT16>
+{
+public:
+  static constexpr float min_range = 0.05f;
+  static constexpr float max_range = 120.0f;
+  static constexpr size_t max_scan_buffer_points = 256000;
```

Comment:
I think this should be `128000` = `16 (channels) * 2 (returns) * 360 (FoV) / 0.009  (highest resolution, if 5 Hz scans are used)`
```suggestion
  static constexpr size_t max_scan_buffer_points = 128000;
```

Reply:
OK